### DISCUSSION TITLE
Data is not populating in external postgres db

### DISCUSSION THREAD

**Discussion by spmishra121** [DependencyTrack/dependency-track#1428] [Q&A]
Hi @stevespringett,

I am facing issue and struck to configure external db in dependency track to use it in production.

I am running DT using docker compose. My external db is installed on some other server, which I have configured in docker-compose.yml file.
When I am starting my docker compose file, it is connecting to that DB properly and relevant tables are created there but when I am creating any data from frontend, it is not reflecting there.

Please help.

Sharing docker compose file...
version: '3.7'

#####################################################
# This Docker Compose file contains two services
#    Dependency-Track API Server
#    Dependency-Track FrontEnd
#####################################################

volumes:
  dependency-track:

services:
  dtrack-apiserver:
    image: dependencytrack/apiserver
    environment:
    # The Dependency-Track container can be configured using any of the
    # available configuration properties defined in:
    # https://docs.dependencytrack.org/getting-started/configuration/
    # All properties are upper case with periods replaced by underscores.
    #
    # Database Properties
     - ALPINE_DATABASE_MODE=external
     - ALPINE_DATABASE_URL=jdbc:postgresql://xx.xx.xx.xx:5432/dependency_track
     - ALPINE_DATABASE_DRIVER=org.postgresql.Driver
     - ALPINE_DATABASE_USERNAME=dtrack
     - ALPINE_DATABASE_PASSWORD=changeme
     - ALPINE_DATABASE_POOL_ENABLED=true
     - ALPINE_DATABASE_POOL_MAX_SIZE=20
     - ALPINE_DATABASE_POOL_MIN_IDLE=10
     - ALPINE_DATABASE_POOL_IDLE_TIMEOUT=300000
     - ALPINE_DATABASE_POOL_MAX_LIFETIME=600000
    #
    # Optional LDAP Properties
    # - ALPINE_LDAP_ENABLED=true
    # - ALPINE_LDAP_SERVER_URL=ldap://ldap.example.com:389
    # - ALPINE_LDAP_BASEDN=dc=example,dc=com
    # - ALPINE_LDAP_SECURITY_AUTH=simple
    # - ALPINE_LDAP_BIND_USERNAME=
    # - ALPINE_LDAP_BIND_PASSWORD=
    # - ALPINE_LDAP_AUTH_USERNAME_FORMAT=%s@example.com
    # - ALPINE_LDAP_ATTRIBUTE_NAME=userPrincipalName
    # - ALPINE_LDAP_ATTRIBUTE_MAIL=mail
    # - ALPINE_LDAP_GROUPS_FILTER=(&(objectClass=group)(objectCategory=Group))
    # - ALPINE_LDAP_USER_GROUPS_FILTER=(member:1.2.840.113556.1.4.1941:={USER_DN})
    # - ALPINE_LDAP_GROUPS_SEARCH_FILTER=(&(objectClass=group)(objectCategory=Group)(cn=*{SEARCH_TERM}*))
    # - ALPINE_LDAP_USERS_SEARCH_FILTER=(&(objectClass=user)(objectCategory=Person)(cn=*{SEARCH_TERM}*))
    # - ALPINE_LDAP_USER_PROVISIONING=false
    # - ALPINE_LDAP_TEAM_SYNCHRONIZATION=false
   # Optional OpenID Connect (OIDC) Properties
    # - ALPINE_OIDC_ENABLED=true
    # - ALPINE_OIDC_ISSUER=https://auth.example.com/auth/realms/example
    # - ALPINE_OIDC_CLIENT_ID=
    # - ALPINE_OIDC_USERNAME_CLAIM=preferred_username
    # - ALPINE_OIDC_TEAMS_CLAIM=groups
    # - ALPINE_OIDC_USER_PROVISIONING=true
    # - ALPINE_OIDC_TEAM_SYNCHRONIZATION=true
    #
    # Optional HTTP Proxy Settings
    # - ALPINE_HTTP_PROXY_ADDRESS=proxy.example.com
    # - ALPINE_HTTP_PROXY_PORT=8888
    # - ALPINE_HTTP_PROXY_USERNAME=
    # - ALPINE_HTTP_PROXY_PASSWORD=
    # - ALPINE_NO_PROXY=
    #
    # Optional Cross-Origin Resource Sharing (CORS) Headers
    # - ALPINE_CORS_ENABLED=true
    # - ALPINE_CORS_ALLOW_ORIGIN=*
    # - ALPINE_CORS_ALLOW_METHODS=GET, POST, PUT, DELETE, OPTIONS
    # - ALPINE_CORS_ALLOW_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *
    # - ALPINE_CORS_EXPOSE_HEADERS=Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count
    # - ALPINE_CORS_ALLOW_CREDENTIALS=true
    # - ALPINE_CORS_MAX_AGE=3600
    deploy:
      resources:
        limits:
          memory: 12288m
        reservations:
          memory: 8192m
      restart_policy:
        condition: on-failure
    ports:
      - '8081:8080'
    volumes:
      - 'dependency-track:/data'
    restart: unless-stopped

  dtrack-frontend:
    image: dependencytrack/frontend
    depends_on:
      - dtrack-apiserver
    environment:
      # The base URL of the API server.
      # NOTE:
      #   * This URL must be reachable by the browsers of your users.
      #   * The frontend container itself does NOT communicate with the API server directly, it just serves static files.
      #   * When deploying to dedicated servers, please use the external IP or domain of the API server.
  - API_BASE_URL=http://xx.xx.xx.xxx:8081
      # - "OIDC_ISSUER="
      # - "OIDC_CLIENT_ID="
      # - "OIDC_SCOPE="
      # - "OIDC_FLOW="
      # - "OIDC_LOGIN_BUTTON_TEXT="
      # volumes:
      # - "/host/path/to/config.json:/app/static/config.json"
    ports:
      - "8080:8080"
    restart: unless-stopped

 **docker-compose log...**

dtrack-apiserver_1  | 2022-03-02 08:34:41,232 INFO [EmbeddedJettyServer] Alpine Executable WAR v1.3.0 (49dc89aa-4f58-404e-b0d6-4001902c5fa1) built on: 2022-02-13T03:18:56Z
dtrack-apiserver_1  | 2022-03-02 08:34:43,291 INFO [Config] --------------------------------------------------------------------------------
dtrack-apiserver_1  | 2022-03-02 08:34:43,293 INFO [Config] OS Name:      Linux
dtrack-apiserver_1  | 2022-03-02 08:34:43,294 INFO [Config] OS Version:   4.14.219-161.340.amzn2.x86_64
dtrack-apiserver_1  | 2022-03-02 08:34:43,294 INFO [Config] OS Arch:      amd64
dtrack-apiserver_1  | 2022-03-02 08:34:43,295 INFO [Config] CPU Cores:    2
dtrack-apiserver_1  | 2022-03-02 08:34:43,309 INFO [Config] Max Memory:   9.6 GB (10,309,074,944.0 bytes)
dtrack-apiserver_1  | 2022-03-02 08:34:43,310 INFO [Config] Java Vendor:  Eclipse Adoptium
dtrack-apiserver_1  | 2022-03-02 08:34:43,313 INFO [Config] Java Version: 11.0.14.1+1
dtrack-apiserver_1  | 2022-03-02 08:34:43,314 INFO [Config] Java Home:    /opt/java/openjdk
dtrack-apiserver_1  | 2022-03-02 08:34:43,314 INFO [Config] Java Temp:    /tmp
dtrack-apiserver_1  | 2022-03-02 08:34:43,314 INFO [Config] User:         dtrack
dtrack-apiserver_1  | 2022-03-02 08:34:43,315 INFO [Config] User Home:    /data/
dtrack-apiserver_1  | 2022-03-02 08:34:43,315 INFO [Config] --------------------------------------------------------------------------------
dtrack-apiserver_1  | 2022-03-02 08:34:43,315 INFO [Config] Initializing Configuration
dtrack-apiserver_1  | 2022-03-02 08:34:43,316 INFO [Config] System property alpine.application.properties not specified
dtrack-apiserver_1  | 2022-03-02 08:34:43,317 INFO [Config] Loading application.properties from classpath
dtrack-apiserver_1  | 2022-03-02 08:34:43,333 INFO [Config] --------------------------------------------------------------------------------
dtrack-apiserver_1  | 2022-03-02 08:34:43,335 INFO [Config] Application:  Dependency-Track
dtrack-apiserver_1  | 2022-03-02 08:34:43,336 INFO [Config] Version:      4.4.1
dtrack-apiserver_1  | 2022-03-02 08:34:43,336 INFO [Config] Built-on:     2022-02-18T16:36:18Z
dtrack-apiserver_1  | 2022-03-02 08:34:43,336 INFO [Config] --------------------------------------------------------------------------------
dtrack-apiserver_1  | 2022-03-02 08:34:43,336 INFO [Config] Framework:    Alpine
dtrack-apiserver_1  | 2022-03-02 08:34:43,337 INFO [Config] Version :     1.10.4
dtrack-apiserver_1  | 2022-03-02 08:34:43,337 INFO [Config] Built-on:     2022-02-13T03:33:58Z
dtrack-apiserver_1  | 2022-03-02 08:34:43,337 INFO [Config] --------------------------------------------------------------------------------

**Comment by stevespringett** [comment]
And do you see the NistDataMirror or the GitHub advisory mirror log messages? Do they complete?

**Reply by spmishra121** [reply]
It is working now :)
